
package paxos;
option cc_generic_services = true;

message PrepareRequest {
    optional uint64 id = 1;
    optional string value = 2;
}

message PrepareResponse {
    optional int32 status = 1;
    optional string value = 2;
}

message AcceptRequest {
    optional string value = 2;
}
message AcceptResponse {
    optional int32 status = 1;
}

service PaxosServer {
    rpc Prepare(PrepareRequest) returns(PrepareResponse);
    rpc Accept(AcceptRequest) returns(AcceptResponse);
}
